calico/felix/endpoint.py

```diff
@@ -315,6 +324,45 @@ def _finish_msg_batch(self, batch, results):
                                                    async=True)
             self._added_to_dispatch_chains = True
 
+        # If changed, report our status back to the datastore.
+        self._maybe_update_status()
+
+    def _maybe_update_status(self):
+        # Work out what status we should report back to the orchestrator.
```

Comment:
I think we shouldn't call `_maybe_update_status` or should return immediately if `REPORT_ENDPOINT_STATUS` is not set; feels cleaner than calculating the status and then not using it.